pyro/generic/generic.py

```diff
@@ -27,6 +31,39 @@ def __getattribute__(self, name):
         return getattr(module, name)
 
 
+def seed(fn=None, rng_key=None):
+    """
+    Seed the callable with the given `rng_key`. Currently, only affects the
+    numpy backend. This can be used as a decorator or context manager.
+
+    :param fn: Python callable. Not required when used as a context manager.
+    :param int rng_key: random number generator seed.
+    """
+    seed_ctx = _SeedCtx(rng_key=rng_key)
+    return seed_ctx if fn is None else seed_ctx(fn)
+
+
+class _SeedCtx(object):
+    def __init__(self, rng_key=None):
+        self.rng_key = rng_key if rng_key is not None else random.randint()
```

Comment:
Could you explain usage to me? Wouldn't this mean that repeated calls to a seeded model would all return the same result? That does not match the current behavior of vanilla Pyro, but maybe I am missing typical usage patterns in NumPyro.

Reply:
> Wouldn't this mean that repeated calls to a seeded model would all return the same result?

That's right. To generate multiple samples from the same model, we'll need to do this seeding in the outer scope.

```
with seed(rng_key=1):
  samples = [model() for _ in range(100)]
```

In NumPyro, we typically do this in a vectorized fashion, but note that a single seeded instance of a model is deterministic.

```
samples = vmap(lambda key: seed(model, key)())(arange(100))
```
.

Reply:
Thanks for explaining, that makes sense!